### DISCUSSION TITLE
LoadDocument now requires a static string collection name

### DISCUSSION THREAD

**Discussion by JoshuaLevett** [ravendb/ravendb#12893] [Q&A] [answered]
I had an index in 4.2 that used a LoadDocument call like this:
```
Map = participationDocs => from participationDoc in participationDocs
let signup = LoadDocument<IEventInstance>(participationDoc.SignupId, participationDoc.SignupId.Substring(0, participationDoc.SignupId.IndexOf('/')))
```

I've updated to 5.2.3 and the index now errors stating that the collection name must be a static string.
In order to accommodate this, I've had to create the multi-map index with selectors like the below (I've only included up to loading the "signup" as the rest isn't relevant)

```
AddMap<SignupParticipation> (participationDocs => from participationDoc in participationDocs
where participationDoc.SignupId.Substring(0, participationDoc.SignupId.IndexOf('/')).ToLower() == "physicalevents"
let signup = LoadDocument<IEventInstance>(participationDoc.SignupId, "physicalevents")
```

```
AddMap<SignupParticipation> (participationDocs => from participationDoc in participationDocs
where participationDoc.SignupId.Substring(0, participationDoc.SignupId.IndexOf('/')).ToLower() == "onlineevents"
let signup = LoadDocument<IEventInstance>(participationDoc.SignupId, "onlineevents")
```

```
AddMap<SignupParticipation> (participationDocs => from participationDoc in participationDocs
where participationDoc.SignupId.Substring(0, participationDoc.SignupId.IndexOf('/')).ToLower() == "ondemands"
let signup = LoadDocument<IEventInstance>(participationDoc.SignupId, "ondemands")
```

This means I have had to write out the map 3 times (more will need to be added too)
Is this a feature that was intentionally removed? Or have I missed a new potential implementation?

Help / advice is appreciated as this may be a pain to maintain

**Comment by ppekrol** [comment, marked as answer]
We added additional validation, your 4.2 index probably was detecting referenced collections incorrectly, hence it was not reacting to changes of these as it should.

The string there must be a const in order for us to detect those referenced collection names.

**Reply by JoshuaLevett** [reply]
Sorry, this is just for my own understanding.
Are you saying that the 4.2 index might not have correctly updated the index if, for example, the "PhysicalEvent" document was updated?

**Reply by ppekrol** [reply]
Yes. It might not react to that change properly.
